sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,203 @@ def transform(self, X):
+    ...       [-1,  1,  3],
+    ...       [ 4,  0, -1],
+    ...       [ 8,  1,  0]
+    ... ])
+    >>> X2 = np.array([
+    ...       [ 5, -1, -1],
+    ...       [-1,  2,  3],
+    ...       [ 2,  4,  0]
+    ... ])
+    >>> indicator = MissingIndicator(missing_values=-1)
+    >>> indicator.fit(X1)
+    MissingIndicator(features='train', missing_values=-1, sparse='auto')
+    >>> X2_tr = indicator.transform(X2)
+    >>> X2_tr
+    array([[0, 1],
+           [1, 0],
+           [0, 0]])
+
+    Attributes
+    ----------
+    feat_with_missing_  : array of shape(n_missing_features, )
+        The features with missing values.
+        Note that this is only stored if features == 'train
+
+    n_features_ : int
+        The number of features during fit time.
+    """
+
+    def __init__(self, missing_values="NaN", features="train", sparse="auto"):
+        self.missing_values = missing_values
+        self.features = features
+        self.sparse = sparse
+
+    def fit(self, X, y=None):
+        """Fit the transformer on X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Input data, where ``n_samples`` is the number of samples and
+            ``n_features`` is the number of features.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        self._validate_params()
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64)
+        self.n_features_ = X.shape[1]
+
+        if (isinstance(self.features, six.string_types) and
+                self.features == "train"):
+            _, self.feat_with_missing_ = self._get_missing_features_info(X)
+
+        return self
+
+    def transform(self, X):
+        """Generate missing values indicator for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        Xt : array or sparse matrix, shape = [n_samples, n_features]
+             The missing indicator for input data
+
+        """
+        if (isinstance(self.features, six.string_types) and
+                self.features == "train"):
+            check_is_fitted(self, "feat_with_missing_", "n_features_")
+        else:
+            check_is_fitted(self, 'n_features_')
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64)
+        if X.shape[1] != self.n_features_:
+            raise ValueError("X has a different number of features "
+                             "than during fitting.")
+
+        imputer_mask, feat_with_missing = self._get_missing_features_info(X)
+
+        if isinstance(self.features, six.string_types):
+            if self.features == "train":
+                features = np.setdiff1d(feat_with_missing,
+                                        self.feat_with_missing_)
+                if features.size > 0:
```

Comment:
Currently it raises a RuntimeWarning. Should I add a parameter to decide whether to ignore or throw a warning ?